server/lib/scanners/baseScanner.ts

```diff
@@ -618,6 +633,56 @@ class BaseScanner<T> {
   get protectedBundleSize(): number {
     return this.bundleSize;
   }
+
+  protected async processUnmonitoredMovie(tmdbId: number): Promise<void> {
+    const mediaRepository = getRepository(Media);
+    await this.asyncLock.dispatch(tmdbId, async () => {
+      const existing = await this.getExisting(tmdbId, MediaType.MOVIE);
+      if (existing && existing.status === MediaStatus.PROCESSING) {
+        existing.status = MediaStatus.UNKNOWN;
+        await mediaRepository.save(existing);
+        this.log(
+          `Movie TMDB ID ${tmdbId} unmonitored from Radarr. Media status set to UNKNOWN.`,
+          'info'
+        );
+      }
+    });
+  }
+
+  protected async processUnmonitoredSeason(
+    tmdbId: number,
+    season: SonarrSeason
+  ): Promise<void> {
+    // Remove unmonitored seasons from Requests
+    const requestRepository = getRepository(MediaRequest);
+    const seasonRequestRepository = getRepository(SeasonRequest);
+
+    const existingRequests = await requestRepository
+      .createQueryBuilder('request')
+      .innerJoinAndSelect('request.media', 'media')
+      .innerJoinAndSelect('request.seasons', 'seasons')
+      .where('media.tmdbId = :tmdbId', { tmdbId: tmdbId })
+      .andWhere('media.mediaType = :mediaType', {
+        mediaType: MediaType.TV,
+      })
+      .andWhere('seasons.seasonNumber = :seasonNumber', {
+        seasonNumber: season.seasonNumber,
+      })
+      .getMany();
+
+    if (existingRequests && existingRequests.length > 0) {
+      existingRequests.forEach((existingRequest) => {
+        existingRequest.seasons.forEach(async (requestedSeason) => {
+          if (requestedSeason.seasonNumber === season.seasonNumber) {
+            this.log(
+              `Removing request for Season ${season.seasonNumber} of tmdbId ${tmdbId} as it is unmonitored`
+            );
+            await seasonRequestRepository.remove(requestedSeason);
+          }
+        });
+      });
```

Comment:
You should use `for` loops here instead of `forEach` if you want the `await` to be effective.